### PR TITLE
Feat: ground combat part 1

### PR DESCRIPTION
**Pull request by felipeStouvenel**
# Feats:

## Finished the refactor for obj_star:

The following scripts were refactored and logic improvements alongside some documentation was added:
- Create_0
- Step_0
- Alarm_0
- Alarm_1
- Alarm_2
- Alarm_3
- Alarm_4
- Draw_0
- Mouse_50

### Next steps for ground combat (in future PRs):

- Create a new obj_planet_maps
- This object will contain some scripts to generate a map on creation of planets
- The first iteration would be to put this into the home system of the player, then once it works implement it to the sector generation.
- This will take some time, but will improve the chapter management and overall planet side gameplay, giving use to scouts and other units. Which would give reason to land on planets and not raid indefinitely.

### CODE REVIEW

File: scripts/scr_star_name/scr_star_name.gml
Comment (GalacticChimp) [inline scripts/scr_star_name/scr_star_name.gml:1]: please split this change out into a separate PR. Will be much easier to review, test, and if we'll have to revert the ground combat feature, we won't have to revert the star name improvements

File: objects/obj_star/Alarm_0.gml [hunk 1]
```diff
@@ -1,200 +1,371 @@
...
+        p_type[3] = "Dead";
+        planet[3] = 1;
+        break;
 }
 
-
-
-
-
-var a,b,c,d,e,f,g,h;
```
Comment (OH296) [inline objects/obj_star/Alarm_0.gml:141]: probably should be some different variable names to a,b,c etc.

File: objects/obj_star/Alarm_1.gml [hunk 2]
```diff
@@ -1,154 +1,157 @@
...
+// the min population of a planet is usually 1/3 of the max. so lava has 1500 max. min is 500. min + random should = max
+// its important to know the population of a planet due to recruitment changing depending on population max
+// If getting to max pop is very rare, it will be awful to recruit from
+// some planets may be better or worse than others depending on their max pop.
+// TODO refactor and improve logic
+for(var i=1; i<=4; i++){
+    p_population[i]=0;// 10B
+    switch (p_type[i]) {
+        case "Lava":
+            p_population[i] = floor(random(1500)) + 500;
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:10]: irandom() instead of floor(random())? Maybe?

File: objects/obj_star/Alarm_1.gml [hunk 3]
```diff
@@ -1,154 +1,157 @@
...
+                system_fleet += 4;
+                break;
+            case "Forge":
+                system_fleet += 8;
+                break;
+            case "Desert":
+            case "Temperate":
+                system_fleet += 1;
+                break;
+            case "Feudal":
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:137]: missing break statement

File: objects/obj_star/Alarm_1.gml [hunk 4]
```diff
@@ -1,154 +1,157 @@
...
+if (owner==2) or (owner==7) or (owner==3){
+    for(var g=1; g<=4; g++){
+        switch (p_type[g]) {
+            case "Hive":
+                system_fleet += 4;
+                break;
+            case "Forge":
+                system_fleet += 8;
+                break;
+            case "Desert":
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:133]: break statement missing

File: objects/obj_star/Alarm_1.gml [hunk 5]
```diff
@@ -160,245 +163,252 @@ if (owner=2) or (owner=7) or (owner=3){// create imperial fleet
...
-    if (p_tau[3]>0){p_owner[3]=8;p_first[3]=8;
-        if (p_type[3]="Forge") or (p_type[3]="Hive") then p_tau[3]=choose(2,3);if (p_type[3]="Ice") then p_tau[3]=choose(1,2);
-        if (p_type[3]="Temperate") or (p_type[3]="Desert") or (p_type[3]="Feudal") then p_tau[3]=choose(3,3,4,4,5);
+    for (var i = 1; i <= 4; i++) {
+        if (p_tau[i] > 0) {
+            p_owner[i] = 8;
+            p_first[i] = 8;
+            
+            switch (p_type[i]) {
+                case "Forge":
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:243]: break statement missing

File: objects/obj_star/Alarm_1.gml [hunk 6]
```diff
@@ -160,245 +163,252 @@ if (owner=2) or (owner=7) or (owner=3){// create imperial fleet
...
+                case "Hive":
+                    p_tau[i] = choose(2,3);
+                    break;
+                case "Ice":
+                    p_tau[i] = choose(1,2);
+                    break;
+                case "Temperate":
+                case "Desert":
+                case "Feudal":
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:252]: also break statements

File: objects/obj_star/Alarm_1.gml [hunk 7]
```diff
@@ -160,245 +163,252 @@ if (owner=2) or (owner=7) or (owner=3){// create imperial fleet
...
-    p_owner[1]=2;p_owner[2]=2;p_owner[3]=2;p_owner[4]=2;
+// Create Nids
+if (owner==9){
+    for (var i = 1; i <= 4; i++) {
+        if (p_population[i] > 0) {
+            p_tyranids[i] = 1;
+            
+            switch (p_type[i]) {
+                case "Forge":
+                case "Hive":
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:280]: break statements

File: objects/obj_star/Alarm_1.gml [hunk 8]
```diff
@@ -502,15 +444,12 @@ if (obj_controller.is_test_map=true){
 // if (obj_controller.is_test_map=true) and (p_owner[2]=1){
```
Comment (OH296) [inline objects/obj_star/Alarm_1.gml:445]: i know chimps been trying to get rid of un-needed comments not sure if this applies here?

File: objects/obj_star/Alarm_4.gml [hunk 9]
```diff
@@ -1,66 +1,70 @@
...
-rando=0;i=0;contin=0;
+// Checks if the planet has industrial facilities to create an ork fleet
+var rando=0,contin=0;
 
-
-repeat(4){
-    i+=1;
+for(var i=1; i<=4; i++){
     // Check for industrial facilities
     if (p_type[i]!="Dead") and (p_type[i]!="Lava"){// Used to not have Ice either
```
Comment (OH296) [inline objects/obj_star/Alarm_4.gml:7]: personally i've found if(!array_contains["dead", "lava", "ice"], p_type[i]){
}
to be a more extendable syntax
but it's all the same really

File: objects/obj_star/Alarm_4.gml [hunk 10]
```diff
@@ -1,66 +1,70 @@
...
+        if (p_orks[i]>=4){
+            var fleet=0;
+            contin=2;
+            if (instance_number(obj_en_fleet)==0) then contin=3;
             if (instance_number(obj_en_fleet)>0) then contin=2;
             
             if (instance_exists(obj_p_fleet)){
-                var ppp;ppp=instance_nearest(x,y,obj_p_fleet);
-                if (point_distance(x,y,ppp.x,ppp.y)<50) and (ppp.action="") then contin=0;
+                var ppp=instance_nearest(x,y,obj_p_fleet);
```
Comment (OH296) [inline objects/obj_star/Alarm_4.gml:16]: variable name ppp could be better

File: objects/obj_star/Alarm_4.gml [hunk 11]
```diff
@@ -1,66 +1,70 @@
...
+                            fleet.escort_number += 1;
+                            break;
+                    }
                     
                     if (fleet.image_index>=5){
                         // eh heh heh
-                        var stue, stue2;stue=0;stue2=0;
-                        var goood;goood=0;
+                        var stue=0,stue2=0;
+                        var goood=0;
```
Comment (OH296) [inline objects/obj_star/Alarm_4.gml:37]: again variable names particularly stue, stue2 could be better

File: objects/obj_star/Create_0.gml [hunk 12]
```diff
@@ -41,7 +47,6 @@ repeat(8){run+=1;
     p_silo[run]=0;
     p_defenses[run]=0;
     p_upgrades[run]="";
```
Comment (OH296) [inline objects/obj_star/Create_0.gml:49]: p_upgrades are now lists so p_upgrades[run]=[];

**Review by OH296 — CHANGES_REQUESTED** [review]
not really sure if i'm doing this well, so don't feel obliged by any of it. Pretty much looks all good to me though.

**Review by OH296 — APPROVED** [review]
looks all good to me

**Pull request merged**